bonus_guides/I_sending_email.md

```diff
@@ -0,0 +1,300 @@
+      key: Application.get_env(:hello_phoenix, :mailgun_key)
+  . . .
+```
+
+When we restart the application and call our `send_welcome_email/1` function, we actually get a response back from mailgun.com telling us our email has been queued.
+
+```console
+iex> HelloPhoenix.Mailer.send_welcome_email("me@example.com")
+{:ok,
+ "{\n  \"id\": \"<20150820050046.numbers.more_numbers@sandbox-our-domain.mailgun.org>\",\n  \"message\": \"Queued. Thank you.\"\n}"}
+```
+
+Great! Time to check our inbox.
+
+Looking at the original source of our email, we can see that it is indeed a multipart email with two parts. The first is our text email, with a Content-Type of "text/plain". The second is our HTML email with a Content-Type of "text/html".
+
+```
+To: them@example.com
+From: Mailgun Sandbox
+ <postmaster@sandbox-our-domain.mailgun.org>
+Subject: Welcome!
+Mime-Version: 1.0
+Content-Type: multipart/alternative; boundary="ab2eaf529cf8442b93154d6e3d98896e"
+
+--ab2eaf529cf8442b93154d6e3d98896e
+Content-Type: text/plain; charset="ascii"
+Mime-Version: 1.0
+Content-Transfer-Encoding: 7bit
+
+Welcome to HelloPhoenix!
+--ab2eaf529cf8442b93154d6e3d98896e
+Content-Type: text/html; charset="ascii"
+Mime-Version: 1.0
+Content-Transfer-Encoding: 7bit
+
+<strong>Welcome to HelloPhoenix Test</strong>
+--ab2eaf529cf8442b93154d6e3d98896e--
+```
+
+# FIXME done up to here
+
+## Rendering Templates
+
+If we're going to render anything other than the simplest HTML while still having a readable `send_welcome_email/1` function, using bare HTML strings is going to present problems. We can use `Phoenix.View.render_to_string/3` to make this manageable.
+
+```elixir
+. . .
+def send_welcome_email(email_address) do
+  . . .
+  html: Phoenix.View.render_to_string(HelloPhoenix.EmailView, "welcome.html", %{})
+  . . .
+end
+```
+
+To make this example work, we'll need a basic `HelloPhoenix.EmailView` defined at `web/views/email_view.ex`.
+
+```elixir
+defmodule HelloPhoenix.EmailView do
+  use HelloPhoenix.Web, :view
+end
+```
+
+We'll also need a new `email` directory in `web/templates` with a `welcome.html.eex` template in it.
+
+```html
+<div class="jumbotron">
+  <h2>Welcome to HelloPhoenix!</h2>
+</div>
+```
+
+# Tidying Up a Bit
+
+# FIXME we can use a private function to specify the welcome text
+
+```elixir
+. . .
+defp welcome_text do
+  """
+  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  """
+end
+. . .
+```
+
+# FIXME if we want to tidy things up a little, we can create a function for html as well
+
+```elixir
+  defp welcome_html
```

Comment:
welcome_email and welcome_text will need to accept some state as passed in arguments